### PR TITLE
Added doxygen and small 'manual'.

### PR DESCRIPTION
**Pull request by lemunozm**

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/40/)

File: README.md [hunk 1]
```diff
@@ -494,6 +495,13 @@ It has, two helper functions to ease the retrieval of all types:
 - `get_all_types`: Retrieves a map with all the defined types.
 - `get_all_scoped_types`: Retrieves a map with all the defined types, whose key is with the scoped name.
 
+### Generator
+Analogous but in opossite direction of `parse()` method, you can generate idl content from ypur defined types.
```
Comment (LuisGP) [inline README.md]:
```suggestion
Analogous but in the opposite direction of the `parse()` method, you can generate IDL content from your defined types.
```

File: README.md [hunk 2]
```diff
@@ -494,6 +495,13 @@ It has, two helper functions to ease the retrieval of all types:
...
+Exists two methods to archive this.
+- `generator(const StructType& type)`, that generates the corresponding IDL for the given structure
+  (and all its dependencies).
+- `genetator(const Module& module)`, that generates the correspondig IDL of the entire module.
```
Comment (LuisGP) [inline README.md]:
```suggestion
- `generator(const Module& module)`, that generates the corresponding IDL of the entire module.
```

**Review by LuisGP — CHANGES_REQUESTED** [review]

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/xtypes-manual-linux/42/)

**Review by LuisGP — APPROVED** [review]

**Pull request merged**